components/VersionSection.tsx

```diff
@@ -56,61 +104,105 @@ export default function VersionSection({
       );
   }, []);
 
+  const statusContent = (
+    <div className="mt-4 flex flex-col items-center space-x-2">
+      {isLoading ? (
+        <>
+          <Loader2 className="h-8 w-8 animate-spin" />
+          <Paragraph variant="smallText" margin="none">
+            Checking for updates...
+          </Paragraph>
+        </>
+      ) : error ? (
+        <>
+          <XCircle className="h-8 w-8 fill-destructive text-destructive-foreground" />
+          <Paragraph
+            className="text-destructive"
+            variant="smallText"
+            margin="none"
+          >
+            There was an error checking for updates.
+          </Paragraph>
+        </>
+      ) : (
+        <>
+          <CheckCircle2 className="h-8 w-8 fill-success text-success-foreground" />
+          <Paragraph variant="smallText" margin="none">
+            You are up to date!
+          </Paragraph>
+        </>
+      )}
+    </div>
+  );
```

Comment:
I wouldn't make this a separate closure - I would have the separate "states" in the render function, and use early return and tighter conditions rather than in a big ternary.